File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::map<int, int> table;
        std::vector<int> res;

        for (int i=0; i<nums.size(); i++) {
            table[nums[i]] = i;
        }

        for (int i=0; i<nums.size(); i++) {
            int diff = target - nums[i];

            if (table[diff] && table[diff] != i) {
                res.push_back(i);
                res.push_back(table[diff]);
                break;
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.go
func twoSum(nums []int, target int) []int {
	front := 0
	rear  := len(nums) - 1

	var res []int

	for front < rear {
		sum := nums[front] + nums[rear]
		if sum == target {
			res = append(res, front+1)
			res = append(res, rear+1)
			break
		} else if sum > target {
			rear--
		} else {
			front++
		}
	}

	return res
}
